src/backend/distributed/commands/cascade_table_operation_for_connected_relations.c

```diff
@@ -623,16 +623,18 @@ ExecuteForeignKeyCreateCommand(const char *commandString, bool skip_validation)
 	 */
 	Assert(IsA(parseTree, AlterTableStmt));
 
+	bool oldValue = SkipConstraintValidation;
+
 	if (skip_validation && IsA(parseTree, AlterTableStmt))
 	{
-		parseTree =
-			SkipForeignKeyValidationIfConstraintIsFkey((AlterTableStmt *) parseTree,
-													   true);
+		SkipConstraintValidation = true;
```

Comment:
this probably messes up the value if an error gets thrown, better to go via set_config_option